### PR TITLE
Board files for LOLIN C3 PICO

### PR DESCRIPTION
**Pull request by tkroo**
Wasn't sure about CIRCUITPY_CREATION_ID so I incremented it by one from the existing lolin_c3_mini ID.
Also, not sure if I need to define something in pins.c for the Lolin I2C connector on the board, which is different from stemma-qt connector.
Successfully builds and flashes on my c3 pico.

### CODE REVIEW

**Review by dhalbert — COMMENTED** [review]
-

**Comment by dhalbert** [comment]
Submit a PR to https://github.com/creationid/creators/blob/main/creations/wemos.md to document the creation ID you chose.

I see that the Lolin I2C connector is not the same :shrug: as QWIIC or STEMMA/QT. That is too bad. You (or was it someone else?) chose 5 and 6 for I2C, but the schematic shows 8 and 10 for the connector. Since the board does not label 5 and 6 for I2C, I'd suggest dropping the `board.I2C()` altogether, to force the user to specify the pins exactly. Or make `board.I2C()` be the connector pins.

File: ports/espressif/boards/lolin_c3_pico/board.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Scott Shawcroft for Adafruit Industries
```
Comment (bill88t) [inline ports/espressif/boards/lolin_c3_pico/board.c:6]: You may want to change this.

**Review by bill88t — APPROVED** [review]

File: ports/espressif/boards/lolin_c3_pico/board.c (hunk 1 above)
Comment (tkroo) [reply ports/espressif/boards/lolin_c3_pico/board.c:6]: I'm hesitant to remove prior copyright notices. What should I change here?

**Comment by tkroo** [comment]
> Submit a PR to https://github.com/creationid/creators/blob/main/creations/wemos.md to document the creation ID you chose.
> 
> I see that the Lolin I2C connector is not the same shrug as QWIIC or STEMMA/QT. That is too bad. You (or was it someone else?) chose 5 and 6 for I2C, but the schematic shows 8 and 10 for the connector. Since the board does not label 5 and 6 for I2C, I'd suggest dropping the `board.I2C()` altogether, to force the user to specify the pins exactly. Or make `board.I2C()` be the connector pins.

I started from the existing lolin_c3_**mini** board files and didn't change the i2c assignments because the board is so similar. Perhaps this needs to be changed for that board also? Anyway, I've reassigned to 8 and 10 to match the board's pinout.

**Comment by dhalbert** [comment]
This seems relevant to the choice of I2C pins: https://github.com/adafruit/circuitpython/pull/5467, and I think explains the current choice. I had not seen this when I wrote my comment above.

@evildave666 @chukwon @wemos @askpatrickw @bill88t tagging you for interest here

**Comment by tkroo** [comment]
> > Submit a PR to https://github.com/creationid/creators/blob/main/creations/wemos.md to document the creation ID you chose.
> > I see that the Lolin I2C connector is not the same shrug as QWIIC or STEMMA/QT. That is too bad. You (or was it someone else?) chose 5 and 6 for I2C, but the schematic shows 8 and 10 for the connector. Since the board does not label 5 and 6 for I2C, I'd suggest dropping the `board.I2C()` altogether, to force the user to specify the pins exactly. Or make `board.I2C()` be the connector pins.
> 
> I started from the existing lolin_c3_**mini** board files and didn't change the i2c assignments because the board is so similar. Perhaps this needs to be changed for that board also? Anyway, I've reassigned to 8 and 10 to match the board's pinout.

Wemos uses SDA = 8 and SCL = 10 in their micropython repository: https://github.com/wemos/micropython/blob/master/ports/esp32/boards/LOLIN_C3_PICO/mpconfigboard.h

File: ports/espressif/boards/lolin_c3_pico/board.c (hunk 1 above)
Comment (tannewt) [reply ports/espressif/boards/lolin_c3_pico/board.c:6]: It's ok to remove. (That's me.)

File: ports/espressif/boards/lolin_c3_pico/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+// From Wemos C3 Mini Schematic
+// https://www.wemos.cc/en/latest/_static/files/sch_c3_mini_v1.0.0.pdf
+// And MP Config
+// https://github.com/micropython/micropython/blob/master/ports/esp32/boards/LOLIN_C3_MINI
+// Status LED
+#define MICROPY_HW_NEOPIXEL         (&pin_GPIO7)
+
+#define CIRCUITPY_BOARD_I2C         (1)
+#define CIRCUITPY_BOARD_I2C_PIN     {{.scl = &pin_GPIO6, .sda = &pin_GPIO5}}
```
Comment (dhalbert) [inline ports/espressif/boards/lolin_c3_pico/mpconfigboard.h:40]: `board.I2C()` should use the same pins as `board.SCL` and `board.SDA`, as on the S2 Pico. So I think you want to change these as below:
```suggestion
#define CIRCUITPY_BOARD_I2C_PIN     {{.scl = &pin_GPIO10, .sda = &pin_GPIO8}}
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
HI - just one thing about I2C.

**Comment by tkroo** [comment]
> HI - just one thing about I2C.

Thanks! updated.

**Review by dhalbert — APPROVED** [review]
OK, looks good. Thanks!

**Pull request merged**